cmds/exp/pxeserver/main.go

```diff
@@ -72,17 +72,35 @@ func (s *dserver4) dhcpHandler(conn net.PacketConn, peer net.Addr, m *dhcpv4.DHC
 		log.Printf("Can't handle type %v", mt)
 		return
 	}
-	if s.mac != nil && !bytes.Equal(m.ClientHWAddr, s.mac) {
-		log.Printf("Not responding to DHCP request for mac %s, which does not match %s", m.ClientHWAddr, s.mac)
-		return
+
+	// In nets with 1 host, and no entries in /etc/hosts, we use the
+	// default command line value for Your IP
+	ip := s.yourIP
+
+	// We preferentially use and /ec/hosts entry if it exists,
+	// with a hostname uab:cd:ef:gh:ij:kl
+	// The /etc/hosts entry takes precedence over all other
+	// options. Since this is dserver4, we force it to be an ip4 address.
+	if i, err := net.LookupIP(fmt.Sprintf("u%s", m.ClientHWAddr)); err == nil {
```

Comment:
can you make a flag for this?

And let users only specify "-your-ip" if -use-etc-hosts flag isn't set?

Reply:
also, v6 support?

Reply:
i'm finding having them both together is very handy. That's the only reason I did not. It may make sense to selectively enable the host name lookup, but ... otoh ... you get a little error message and that's it. 

One thought is that to disable anonymous dhcp, the yourIP could be default ""

Then if yourIP is "", it means "always use /etc/hosts", and if it is an IP, it means "never use /etc/hosts"

except, sometimes ... having both is handy. I dunno.

Reply:
but definitely if they are mutually exclusive, it should be one flag, I guess? so yourIP could be that flag.

Reply:
actually it's also not just /etc/hosts. I'm using net.LookupIP. It can be anything. It's just most commonly /etc/hosts.